processing/src/test/java/org/apache/druid/query/aggregation/SerializablePairLongStringComplexMetricSerdeTest.java

```diff
@@ -69,7 +69,7 @@ public void testLargeString() throws Exception
     assertExpected(ImmutableList.of(new SerializablePairLongString(
         100L,
         randomStringUtils.randomAlphanumeric(2 * 1024 * 1024)
-    )), 2097182, 2103140);
+    )), 2097182, 2103139);
```

Comment:
What is happening here? It seems the `expectedCompressedSize` is changing- what drives this, and is it backwards compatible with older readers and writers?

Reply:
I was also worried when I've first saw these tests breaking :)
but its due to the fact that the random algorithm was altered in `commons-lang3:3.15.0`.
The reason that matters is that `RandomStringUtils` uses `org.apache.commons.lang3.RandomStringUtils` internally.

Prior to `<3.15.0` these tests pass with all the other dep updates and require these updates afterwards.

https://commons.apache.org/proper/commons-lang/changes-report.html#a3.15.0
```
Reimplement RandomUtils and RandomStringUtils on top of SecureRandom#getInstanceStrong() #1235. Thanks to Gary Gregory, Henri Yandell, Fabrice Benhamouda.
```

Reply:
ah, thank you for looking into this.